### PR TITLE
Adding Event Index Sequence checking to UTs

### PR DESCRIPTION
**Pull request by Tonix517**
Adding new checking in UT cases to make sure numbers in event index sequence are unique and increasing. As a follow-up to https://github.com/onflow/flow-go/pull/3371#pullrequestreview-1139592903

changes to FVM and execution computer UT are split into separate commits.

### CODE REVIEW

File: engine/execution/computation/computer/computer_test.go [hunk 1]
```diff
@@ -947,3 +952,15 @@ func generateEvents(eventCount int, txIndex uint32) []flow.Event {
 	}
 	return events
 }
+
+func ensureEventsIndexSeq(t *testing.T, events []flow.Event, chainID flow.ChainID) {
+	expectedEventIndex := uint32(0)
+	for _, event := range events {
+		require.Equal(t, expectedEventIndex, event.EventIndex)
+		if testutil.IsServiceEvent(event, chainID) {
+			expectedEventIndex += 2
```
Comment (Tonix517) [inline engine/execution/computation/computer/computer_test.go:961]: In our code at https://github.com/onflow/flow-go/blob/master/fvm/environment/event_emitter.go#L154-L155, for service events we do double counting (original behavior), so the event after one service event will have index as `+2`. 
However if `AppendServiceEvent()` fails and `AppendEvent()` is skipped, then if `EmitEvent()` is called again, we can continue to have unique index values. If this use case can happen, we may want to keep the existing counting logic.

Comment (janezpodhostnik) [reply engine/execution/computation/computer/computer_test.go:961]: This looks like a bug. Can you open an issue for it?

**Review by janezpodhostnik — APPROVED** [review]
Nice!

File: fvm/fvm_fuzz_test.go [hunk 2]
```diff
@@ -307,3 +310,25 @@ func bootstrapFuzzStateAndTxContext(tb testing.TB) (bootstrappedVmTest, transact
 			chain:        bootstrappedVMTest.chain,
 		}
 }
+
+func ensureEventsIndexSeq(t *testing.T, events []flow.Event, chainID flow.ChainID) {
```
Comment (pattyshack) [inline fvm/fvm_fuzz_test.go:314]: you're duplicating these two functions.  maybe add it to a shared testutil

**Review by pattyshack — APPROVED** [review]
see my duplicated code comment, but otherwise i don't have issues with this

File: engine/execution/computation/computer/computer_test.go (hunk 1 above)
Comment (Tonix517) [reply engine/execution/computation/computer/computer_test.go:961]: filed #3393

**Comment by Tonix517** [comment]
bors merge

**Comment by Tonix517** [comment]
will merge it first and treat https://github.com/onflow/flow-go/issues/3393 as a separate fix in future PR

**Pull request merged**